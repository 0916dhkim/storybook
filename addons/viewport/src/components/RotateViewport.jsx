import React, { Component } from 'react';
import PropTypes from 'prop-types';
import * as styles from './styles';

export class RotateViewport extends Component {
  static propTypes = {
    disabled: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
    active: PropTypes.bool,
  };

  render() {
    const { active, ...props } = this.props;

    const disabledStyles = props.disabled ? styles.disabled : {};
    const actionStyles = {
      ...styles.action,
      ...disabledStyles,
    };

    return (
      <div style={styles.row}>
        <label style={styles.label}>Rotate</label>

        <button {...props} style={actionStyles}>
          {active ? 'Vertical' : 'Landscape'}
        </button>
      </div>
    );
  }
}
