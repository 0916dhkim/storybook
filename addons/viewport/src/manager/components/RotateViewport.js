import React from 'react';
import PropTypes from 'prop-types';

import { Label, Row, Button } from './styles';

export const RotateViewport = ({ active, ...props }) => (
  <Row>
    <Label htmlFor="rotate">Rotate</Label>
    <Button id="rotate" {...props}>
      {active ? 'Vertical' : 'Landscape'}
    </Button>
  </Row>
);

RotateViewport.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool,
};

RotateViewport.defaultProps = {
  disabled: true,
  active: false,
};
