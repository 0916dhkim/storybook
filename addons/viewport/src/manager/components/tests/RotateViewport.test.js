import React from 'react';
import { shallow } from 'enzyme';
import { RotateViewport } from '../RotateViewport';

const setup = addition => {
  const props = {
    onClick: jest.fn(),
    disabled: false,
    ...addition,
  };

  const result = shallow(<RotateViewport {...props} />);

  return { result, props };
};

describe('Viewport/RotateViewport', () => {
  it('renders correctly', () => {
    const { result } = setup();
    expect(result).toMatchSnapshot();
  });

  it('has a click handler set via props', () => {
    const { result, props } = setup();
    const button = result.find('Styled(button)');
    expect(button).toHaveProp('onClick');

    button.simulate('click');
    expect(props.onClick).toHaveBeenCalled();
  });

  it('renders the correctly if not-disabled', () => {
    const { result } = setup({ disabled: false });
    expect(result.find('Styled(button)')).toHaveProp('disabled', false);
  });

  it('renders the correctly if disabled', () => {
    const { result } = setup({ disabled: true });
    expect(result.find('Styled(button)')).toHaveProp('disabled', true);
  });

  it('renders the correctly if not-active', () => {
    const { result } = setup({ active: false });
    expect(result.html()).toContain('Landscape');
  });

  it('renders the correctly if active', () => {
    const { result } = setup({ active: true });
    expect(result.html()).toContain('Vertical');
  });
});
