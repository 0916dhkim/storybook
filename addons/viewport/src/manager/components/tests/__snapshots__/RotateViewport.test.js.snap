// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viewport/RotateViewport renders correctly 1`] = `
<Styled(div)>
  <Styled(label)
    htmlFor="rotate"
  >
    Rotate
  </Styled(label)>
  <Styled(button)
    disabled={false}
    id="rotate"
    onClick={[MockFunction]}
  >
    Landscape
  </Styled(button)>
</Styled(div)>
`;
